erts/etc/common/erlexec.c

```diff
@@ -763,6 +764,19 @@ int main(int argc, char **argv)
 		    }
 		    break;
 
+		  case 'p':
+		    if (strcmp(argv[i], "-proto_dist") == 0) {
+			if (i+1 >= argc)
```

Comment:
For this check, I've reused the same code as found for other similar checks in the rest of the file, however they are not working as intended currently, cf. issue https://github.com/erlang/otp/issues/4624

Reply:
I've switched the previous check to use the new NEXT_ARG_CHECK(); introduced in 022cea1